src/index.js

```diff
@@ -0,0 +1,167 @@
+/*
+  MIT License http://www.opensource.org/licenses/mit-license.php
+  Author Tobias Koppers @sokra
+*/
+
+import validate from '@webpack-contrib/schema-utils';
+import {
+  getOptions,
+  getRemainingRequest,
+  getCurrentRequest,
+  stringifyRequest,
+} from 'loader-utils';
+import postcss from 'postcss';
+
+import schema from './options.json';
+import plugin from './plugin';
+import SyntaxError from './SyntaxError';
+
+let runtimeFile = require.resolve('./runtime');
+
+try {
+  // run test
+  runtimeFile = require.resolve('../dist/runtime');
+} catch (e) {} // eslint-disable-line no-empty
+
+export default function loader(content, map, meta) {
+  const options = getOptions(this) || {};
+
+  validate({ name: 'CSS Loader', schema, target: options });
+
+  const cb = this.async();
+  const { url, import: importOpt, sourceMap, importLoaders } = Object.assign(
+    {},
+    { url: true, import: true, sourceMap: false, importLoaders: 0 },
+    options
+  );
+
+  let prevMap = map;
+
+  if (sourceMap && prevMap) {
+    if (typeof prevMap === 'string') {
+      prevMap = JSON.parse(prevMap);
+    }
+
+    if (prevMap.sources) {
+      prevMap.sources = prevMap.sources.map((source) =>
+        source.replace(/\\/g, '/')
+      );
+      prevMap.sourceRoot = '';
+    }
+  } else {
+    // Some loaders (example `"postcss-loader": "1.x.x"`) always generates source map, we should remove it
+    prevMap = null;
+  }
+
+  let contentOrAst = content;
+
+  // Reuse CSS AST (PostCSS AST e.g 'postcss-loader') to avoid reparsing
+  if (meta) {
+    const { ast } = meta;
+
+    if (ast && ast.type === 'postcss') {
+      contentOrAst = ast.root;
+    }
+  }
+
+  const plugins = [
+    plugin({
+      loaderContext: this,
+      url,
+      import: importOpt,
+      importLoaders,
+    }),
+  ];
+
+  const postcssOptions = {
+    // We need a prefix to avoid path rewriting of PostCSS
+    from: `/css-loader!${getRemainingRequest(this)
```

Comment:
@michael-ciniawsky it is break source maps with `sass-loader`/less-loader`, we have a lot of problems with source map and let's solve this in other PR (`css-loader@2.1.0` or other version). This PR remove `css` modules and other unnecessary option and don't touch source maps, because now they working in 99% cases.

Reply:
Ok, but I thought `getRemainingRequest.pop... === this.resourcePath`, the `/css-loader!` prefix should be kept for now

Reply:
@michael-ciniawsky `getRemainingRequest !== this.resourcePath` when you use `sass-loader/less-loader`